packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -85,20 +104,30 @@ class DefaultMaterialLocalizations implements MaterialLocalizations {
   ///
   /// [LocalizationsDelegate] implementations typically call the static [load]
   /// function, rather than constructing this class directly.
-  DefaultMaterialLocalizations(this.locale) {
+  factory DefaultMaterialLocalizations(Locale locale) {
     assert(locale != null);
-    _nameToValue = localizations[locale.toString()]
-      ?? localizations[locale.languageCode]
-      ?? localizations['en']
-      ?? <String, String>{};
+    final Map<String, String> englishLocalizations = localizations['en'];
+    assert(englishLocalizations != null);
+
+    final Map<String, String> result = <String, String>{};
+    result.addAll(englishLocalizations);
+    if (locale.languageCode != 'en' && localizations.containsKey(locale.languageCode)) {
+      result.addAll(localizations[locale.languageCode]);
+    }
+    if (localizations.containsKey(locale.toString())) {
+      result.addAll(localizations[locale.toString()]);
+    }
+    return new DefaultMaterialLocalizations._(locale, result);
```

Comment:
Why are we mixing English into the translations?

Reply:
To use English text rather than `null` for keys that are missing in the `.arb` file corresponding to `locale`. Let me know what you'd like to do instead.

Reply:
Just don't have the keys missing in the locale. :-)

Reply:
Done.